template/app/api/chains/supported/route.ts

```diff
@@ -0,0 +1,20 @@
+import { NextResponse } from 'next/server';
+import { getChainsForEnvironment } from '../../../../src/store/supportedChains';
+
+/**
+ * Handler for the /api/chains/supported route, this route will return all the supported
+ * chains for this application.
+ * @param req
+ * @param res
+ */
+export async function GET(): Promise<Response> {
+  try {
+    const chains = getChainsForEnvironment();
+    return NextResponse.json(chains, { status: 200 });
+  } catch (error) {
+    console.error('Error fetching chains:', error);
+    return NextResponse.json({}, { status: 500, statusText: 'Internal Server Error' });
+  }
+}
+
+export const dynamic = 'force-dynamic';
```

Comment:
@rob.polak I was looking through the code, I couldnt find where we were using nextURL.searchParams for /api/chains/supported. Maybe we could remove this line for this route? 
I am suggesting this because I think the supported chains would not change, so we might get some caching benefits. What are your thoughts?

Reply:
@robpolak @Sneh1999 I think the 'force-dynamic' can also relate to caching, so if the getChainsForEnvironment is considered dynamic (will return something new each time) then we would want to keep this dynamic

Reply:
I think we should have caching on this route as the list of supported chains currently doesnt change a lot, open to thoughts!

Reply:
Side note, we should document in some way how force-dynamic works, or at least have link for public docs.